lib/dtutils.lua

```diff
@@ -0,0 +1,866 @@
+--[[
+
+    dtutils.lua - common darktable lua functions
+
+    Copyright (C) 2016 Bill Ferguson <wpferguson@gmail.com>.
+    Copyright (C) 2016 Tobias Jakobs
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+dtutils = {}
+
+local dt = require "darktable"
+
+local log = require "lib/libLog"
+
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{3,0,0})
+
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("dtutils",dt.configuration.config_dir.."/lua/")
```

Comment:
Okay, I'm a little smarter about gettext ;).  IMO this should always point to dt.configuration.config_dir .. "/lua/locale".  This is where the "compiled" .po files should end up, and where gettext should look for them.  I built tools/gen_mo.lua to find all the .po files, compile them, and put them in the correct place under locale so that the user doesn't have to.